src/solid_poro_3D_ele/4C_solid_poro_3D_ele_calc_lib.hpp

```diff
@@ -66,6 +73,126 @@ namespace Discret::Elements::Internal
 
 namespace Discret::Elements
 {
+  enum class PorosityFormulation : std::uint8_t
+  {
+    implicit_porosity,
+    explicit_porosity
+  };
```

Comment:
Here I'm unsure.

With `implicit_porosity`, I refer to the case where the porosity is calculated by the material, and `explicit_porosity` is where it is explicitly solved for the porosity as a primary variable. Not sure whether this is clear.

Reply:
```suggestion
  enum class PorosityFormulation : std::uint8_t
  {
    porosity_from_material_law,
    porosity_as_primary_variable
  };
```
something like that would be more intuitive for @bugrahantemur and me :)

Reply:
I like the class variables names `porosity_from_material_law` and `porosity_as_primary_variable`, however I think it would be good to change the enum to either only `Formulation` (if we have a proper namespace before it (fex: `PoroElast::Formulation`)), but I think this is not the case, so I would actually suggest to changing it to `PoroElastFormulation`. Then it should be immedialty clear what is meant.